Now tracking new test failures

fishygeek91 opened the pull request:
Added 3 new fields to help with this. 
 **Test_Run**: Failure_Summary__c (test field that summarizes new failures) and New_Test_Failures__c (count rollup of new field below).
**Test_Run_Method_Result**: New_Failure__c(boolean)

TestRunProcessor has been changed to update Failure_Summary__c, which is now included in the Test_Run_Notifications.email (template) along with New_Failures__c.

fishygeek91 committed:
- now tracking new test failures and improved messaging to include tests/methods/stacktrace

ChuckJonas commented on force-app/main/default/classes/TestRunProcessor.cls:
```diff
@@ -133,4 +138,28 @@ public with sharing class TestRunProcessor implements Schedulable {
+        }
+        if (!failureMessages.isEmpty()) {
+            String failureSummary = String.join(failureMessages, '\n');
+            testRun.Failure_Summary__c = failureSummary;
```
Maybe this should be called "New_Failure_Summary"?  Also, we might further decompose this and make it easier to manage by having the individual "Failure Summary" as a formula on the `Test_Run_Method_Result__c`.

That makes it a bit easier to make tweaks to and a bit more flexible in allow us to use it in other ways

ChuckJonas replied:
OR... We refactor the Email template to be a [VF page](https://developer.salesforce.com/docs/atlas.en-us.pages.meta/pages/pages_email_templates_intro.htm) and then we don't even need this field... I actually like that approach better as it allows us to have much more templating control.  Would be good exposure for you to VF pages.

ChuckJonas commented on force-app/main/default/objects/Test_Run_Method_Result__c/fields/New_Failure__c.field-meta.xml:
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<CustomField xmlns="http://soap.sforce.com/2006/04/metadata">
+    <fullName>New_Failure__c</fullName>
```
this could actually just be a formula just checking if `ISBLANK(First_Failure__c)`?

ChuckJonas replied:
however, if we do it that way, you may not be able to use it in the rollup for the Test_Run

fishygeek91 committed:
- added a VF email template
- added Runtime to report
- added new failures
- email alert now includes relavent test failure info

fishygeek91 committed:
- updated to use a map in controller to group by class

fishygeek91 committed:
-added new object and fixed failure audit

fishygeek91 committed:
-updated workflow to use new template

fishygeek91 committed:
-getting workflow to correct folder

fishygeek91 committed:
-deleted old email template

fishygeek91 committed:
-updated install link in readme
- updated version numbers

fishygeek91 committed:
- added unit test for email controller
